### PR TITLE
docs: add more content around database connections

### PR DESCRIPTION
**Pull request by thomasheartman**
Improve database connection configuration documentation (#1445)

### CODE REVIEW

File: website/docs/deploy/configuring-unleash.md [hunk 1]
```diff
@@ -203,7 +203,82 @@ function getLogger(name) {
...
+| `pool.max`               | `DATABASE_POOL_MAX`             | 4              | The maximum number of connections in the connection pool.                                                                                                                                |
+| `pool.idleTimeoutMillis` | `DATABASE_POOL_IDLE_TIMEOUT_MS` | 30000          | The amount of time (in milliseconds) that a connection must be idle for before it is marked as a candidate for eviction.                                                                 |
+|                          | `DATABASE_URL`                  | N/A            | A string that matches the [libpq connection string](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING), such as `postgres://USER:PASSWORD@HOST:PORT/DATABASE`. |
+|                          | `DATABASE_URL_FILE`             | N/A            | The path to a file that contains a [libpq connection string](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING).                                               |
```
Comment (thomasheartman) [inline website/docs/deploy/configuring-unleash.md:205]: Are these supported in JS or are they env var only?

Comment (sighphyre) [reply website/docs/deploy/configuring-unleash.md:205]: They're in the JS as well, DATABASE_URL maps to databaseUrlFile and DATABASE_URL_FILE maps to databaseUrlFile, both in the top level options so:

const unleashOptions = {
  databaseUrlFile: "something",
  databaseUrl: "something-else"
  db: {
    user: 'unleash_user',
    password: 'passord',
   } 
}
 
Both env vars are ignored if their JS equivalent option is set in the config

**Review by sighphyre — APPROVED** [review]
Nice!

File: website/docs/deploy/configuring-unleash.md (hunk 1 above)
Comment (thomasheartman) [reply website/docs/deploy/configuring-unleash.md:205]: Thanks! I'm assuming you meant that `DATABASE_URL` maps to `databaseUrl` (as per your example) and not `databaseUrlFile`. 

Also a great point that in-code config overrides env vars: I'll make sure to specify that!

Comment (thomasheartman) [reply website/docs/deploy/configuring-unleash.md]: @sighphyre Just to confirm: I've assumed that in-code config _always_ overrides env vars if they're both present. Is that correct or is that just for db url and url file?

Comment (sighphyre) [reply website/docs/deploy/configuring-unleash.md]: @thomasheartman It should be, yeah! There's no global force override for this so it's done on a case by case basis but I can't find a case where this isn't true

**Pull request merged**